cmd/deploy.go

```diff
@@ -530,3 +518,30 @@ func (dc *deployCmd) configureContainerMonitoringAddon(ctx context.Context, k8sC
 	}
 	return nil
 }
+
+//get domain of azure log analytics workspace based on the cloud or azure stack dependenciesLocation
+func (dc *deployCmd) getLogAnalyticsWorkspaceDomain() string {
+	var workspaceDomain string
+	var cloudOrDependenciesLocation string
+	cloudOrDependenciesLocation = dc.containerService.GetCloudSpecConfig().CloudName
+	if dc.containerService.Properties.IsAzureStackCloud() {
+		cloudOrDependenciesLocation = string(dc.containerService.Properties.CustomCloudProfile.DependenciesLocation)
+	}
+	switch cloudOrDependenciesLocation {
```

Comment:
I'm not sure this switch statement does what you think it does. For example, `workspaceDomain` will be `""` if `"AzureGermanCloud"` is the value of `cloudOrDependenciesLocation`.

See: https://play.golang.org/p/WkA59EajxGP

Reply:
> I'm not sure this switch statement does what you think it does. For example, `workspaceDomain` will be `""` if `"AzureGermanCloud"` is the value of `cloudOrDependenciesLocation`.
> 
> See: https://play.golang.org/p/WkA59EajxGP
Thanks @devigned  for catching this issue. Yes, your right.  interesting bug in my code. Fixed.